### PR TITLE
Enhancement/adding network availability check

### PR DESCRIPTION
**Pull request by sanjaymahto**

### CODE REVIEW

**Comment by sanjaymahto** [comment]
> @googlebot I signed it!

File: network/index.js [hunk 1]
```diff
@@ -37,9 +37,10 @@ const useNetworkStatus = initialEffectiveConnectionType => {
   useEffect(() => {
     if (!unsupported) {
       const navigatorConnection = navigator.connection;
+      const networkAvailability = navigator.onLine;
```
Comment (mlampedx) [inline network/index.js]: Perhaps you could add event listeners for the `online` and `offline` events as well, so this hook can stay updated as the user's network connectivity changes throughout their session?

File: network/network.test.js [hunk 2]
```diff
@@ -113,6 +113,19 @@ describe('useNetworkStatus', () => {
     expect(result.current.effectiveConnectionType).toEqual('4g');
   });
 
+  test('should update the effectiveConnectionType state when network get disabled', () => {
```
Comment (mlampedx) [inline network/network.test.js]: It's probably worth explicitly testing defining `navigator.onLine` with `true/false` states prior to calling `renderHook` as well.

File: network/network.test.js [hunk 3]
```diff
@@ -113,6 +113,19 @@ describe('useNetworkStatus', () => {
...
+    global.navigator.connection = {
+      ...ectStatusListeners,
+      effectiveType: '2g'
+    };
+
+    const { result } = renderHook(() => useNetworkStatus());
+    global.navigator.connection.effectiveType = 'network-unavailable';
```
Comment (mlampedx) [inline network/network.test.js]: Is this `effectiveType` officially supported in a browser? I don't see it in [the spec](http://wicg.github.io/netinfo/) 🤔

Comment (stramel) [reply network/network.test.js]: I don't believe so, I haven't encountered this before. Personally, I would prefer to stick closer to the spec

File: network/index.js (hunk 1 above)
Comment (sanjaymahto) [reply network/index.js]: Yeah, sure. that would be a better way.

File: network/network.test.js (hunk 3 above)
Comment (addyosmani) [reply network/network.test.js]: This is the first time I've encountered `network-unavailable` and wasn't aware it was supported. Is it possible to learn where this value came from? :)

Comment (sanjaymahto) [reply network/network.test.js]: Yeah, It's not supported. Sorry!, My bad. I've refactored the code.

**Pull request closed without merging**